### DISCUSSION TITLE
Could run in Linux arm64 with drm and glesv2.so?

### DISCUSSION THREAD

**Discussion by Coloryr** [AvaloniaUI/Avalonia#14746] [Q&A]
I want my program to run on Linux arm64 using DRM mode, but the GPU driver is only glesv2. So, can I call GPU rendering
I previously tested that as long as the software starts, it will make a core run full and the frame rate is very low on my development board

**Comment by tbibre** [comment]
I've been able to make it works, but there is maybe some work to do on your distro to enable hardware acceleration.

Here is the code I have in Program.cs, there are some tests with differents arguments as I was trying to make the hardware acceleration works:


        public static void Main(string[] args)
        {
            var Builder = BuildAvaloniaApp();

            if (args.Contains("--fb") )
            {
                SilenceConsole();           
                Builder.StartLinuxFbDev(args);
            }
            else
            {
                if (args.Contains("--drm0"))
                {
                    SilenceConsole();
                    Builder.StartLinuxDrm(args, card: "/dev/dri/card0", 1D);
                }
                else
                {
                    if (args.Contains("--drm1"))
                    {
                        SilenceConsole();
                        Builder.StartLinuxDrm(args, card: "/dev/dri/card1", scaling: 1);
                    }
                    else
                    {
                        Builder.StartWithClassicDesktopLifetime(args);
                    }
                }
            }
        }

        private static void SilenceConsole()
        {
            new Thread(() =>
            {
                Console.CursorVisible = false;
                while (true)
                {
                    Console.ReadKey(true);
                }
            })
            { IsBackground = true }.Start();
        }

        // Avalonia configuration, don't remove; also used by visual designer.
        public static AppBuilder BuildAvaloniaApp()
            => AppBuilder.Configure<App>()
                .UsePlatformDetect()
                .LogToTrace()
                .With(new SkiaOptions
                {
                    MaxGpuResourceSizeBytes = 512000000,
                })
                .With(new LinuxFramebufferPlatformOptions
                {
                    Fps = 30
                });

**Reply by Coloryr** [reply]
It's will automatically use GLES?

**Reply by tbibre** [reply]
It will use what is setup in your distro.

**Reply by Coloryr** [reply]
Could be know if it uses GLES?
